replicate state_dict tests

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90868

Simple tests for replicate() state_dict. Ensuring composition with
FSDP works will come as a follow up.

Differential Revision: [D42048131](https://our.internmc.facebook.com/intern/diff/D42048131/)

rohan-varma committed:
replicate state_dict tests

Simple tests for replicate() state_dict. Ensuring composition with
FSDP works will come as a follow up.

Differential Revision: [D42048131](https://our.internmc.facebook.com/intern/diff/D42048131/)

[ghstack-poisoned]

rohan-varma committed:
Update on "replicate state_dict tests"


Simple tests for replicate() state_dict. Ensuring composition with
FSDP works will come as a follow up.

Differential Revision: [D42048131](https://our.internmc.facebook.com/intern/diff/D42048131/)

[ghstack-poisoned]

awgu commented on test/distributed/_composable/test_replicate.py:
```diff
@@ -26,6 +26,43 @@ def forward(self, x):
+            pass
+
+    def _check_state_dict_parity(self, sd_1, sd_2):
+        for k1, k2 in zip(sd_1.keys(), sd_2.keys()):
```
nit: No need to change but no need for `keys()`.

awgu commented on test/distributed/_composable/test_replicate.py:
```diff
@@ -26,6 +26,43 @@ def forward(self, x):
+            self.assertEqual(v1, v2)
+
+    def test_replicate_single_module_save_load(self):
+        pass
```
nit: Should we remove this `pass`?

awgu reviewed (approved):
LGTM!

rohan-varma committed:
Update on "replicate state_dict tests"


Simple tests for replicate() state_dict. Ensuring composition with
FSDP works will come as a follow up.

Differential Revision: [D42048131](https://our.internmc.facebook.com/intern/diff/D42048131/)

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge -f "CI passed"
